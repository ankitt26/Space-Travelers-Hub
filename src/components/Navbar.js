import { NavLink } from 'react-router-dom';
import planet from '../assets/planet.png';
import styles from '../CSS/Navbar.module.css';

const Navbar = () => (
  <header className={styles.header}>
    <img src={planet} alt="Logo" className={styles.logo} />
    <h1 className={styles.title}>Space Travellers&apos; Hub</h1>
    <nav className={styles.navigation}>
      <ul className={styles.nav_list}>
        <li>
          <NavLink className={styles.link} to="/">Rockets</NavLink>
        </li>
        <li>
          <NavLink className={styles.link} to="/Missions">Missions</NavLink>
        </li>
        <li className={styles.br}>|</li>
        <li>
          <NavLink className={styles.link} to="/MyProfile">My Profile</NavLink>
        </li>
      </ul>
    </nav>
  </header>

);

export default Navbar;
