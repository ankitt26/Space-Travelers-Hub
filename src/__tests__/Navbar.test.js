import renderer from "react-test-renderer";
import { Provider } from "react-redux";
import Navbar from '../components/Navbar'
import Store from "../Redux/store";
import { MemoryRouter } from "react-router";

it("MyProfile component test", () => {
  const tree = renderer
    .create(
      <MemoryRouter>
        <Provider store={Store}>
          <Navbar />
        </Provider>
      </MemoryRouter>
    )
    .toJSON();
  expect(tree).toMatchSnapshot();
});

