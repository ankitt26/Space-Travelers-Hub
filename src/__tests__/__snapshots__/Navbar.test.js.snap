// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`MyProfile component test 1`] = `
<header
  className="header"
>
  <img
    alt="Logo"
    className="logo"
    src="planet.png"
  />
  <h1
    className="title"
  >
    Space Travellers' Hub
  </h1>
  <nav
    className="navigation"
  >
    <ul
      className="nav_list"
    >
      <li>
        <a
          aria-current="page"
          className="link active"
          href="/"
          onClick={[Function]}
        >
          Rockets
        </a>
      </li>
      <li>
        <a
          className="link"
          href="/Missions"
          onClick={[Function]}
        >
          Missions
        </a>
      </li>
      <li
        className="br"
      >
        |
      </li>
      <li>
        <a
          className="link"
          href="/MyProfile"
          onClick={[Function]}
        >
          My Profile
        </a>
      </li>
    </ul>
  </nav>
</header>
`;
